nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-core/src/main/java/org/nuxeo/ecm/directory/AbstractDirectory.java

```diff
@@ -360,4 +404,47 @@ protected void fallbackOnDefaultCache() {
             }
         }
     }
+
+    /**
+     * Consumer to load data from CSV according to the dataLoadingPolicy.
+     */
+    protected static class CSVLoaderConsumer implements Consumer<Map<String, Object>> {
+
+        protected final String dataLoadingPolicy;
+
+        protected final Session session;
+
+        protected final String schema;
+
+        public CSVLoaderConsumer(String dataLoadingPolicy, Session session, String schema) {
+            this.dataLoadingPolicy = dataLoadingPolicy;
+            this.session = session;
+            this.schema = schema;
+        }
+
+        @Override
+        public void accept(Map<String, Object> fieldMap) {
+            if (DATA_LOADING_POLICY_REJECT_DUPLICATE.equals(dataLoadingPolicy)) {
+                // leverage DB constraints
+                ((BaseSession) session).createEntryWithoutReferences(fieldMap);
+            } else {
+                Object rawId = fieldMap.get(session.getIdField());
+                if (rawId == null) {
+                    throw new DirectoryException("A line is missing the entry id", SC_BAD_REQUEST);
```

Comment:
Shouldn't we log the `fieldMap` to aid debugging?

Reply:
I put a try/catch/addInfo/re-throw bloc in `DirectoryCSV` class in order to add as info the line number. We did that to avoid printing in logs sensible data.

While we're doing this change, I saw in `DirectoryCSV` we're printing csv lines when record is non consistent, I remove this part of log.